gnucash/register/register-gnome/completioncell-gnome.c

```diff
@@ -0,0 +1,1010 @@
+    {
+        gnc_item_list_set_sort_enabled (box->item_list,
+                                        FALSE, box->sort_column_id);
+    }
+
+    box->stop_searching = FALSE;
+    box->using_anywhere = FALSE;
+    gtk_list_store_clear (box->item_store);
+
+    if (box->sort_enabled) // if sorting, enable it
+    {
+        gnc_item_list_set_sort_enabled (box->item_list,
+                                        TRUE, box->sort_column_id);
+    }
+
+    unblock_list_signals (cell);
+
+    // reconect list store to tree view
+    gnc_item_list_connect_store (box->item_list, store);
+
+    gnc_item_edit_hide_popup (box->item_edit);
+    box->list_popped = FALSE;
+    box->occurrence_id = 0;
+}
+
+void
+gnc_completion_cell_clear_menu (CompletionCell* cell)
+{
+    PopBox* box;
+
+    if (cell == NULL)
+        return;
+
+    box = cell->cell.gui_private;
+    if (box == NULL)
+        return;
+
+    if (box->item_list != NULL)
+    {
+        g_hash_table_remove_all (box->item_hash);
+        item_store_clear (cell);
+    }
+}
+
+void
+gnc_completion_cell_add_menu_item (CompletionCell* cell,
+                                   const char* menustr)
+{
+    PopBox* box;
+
+    if (cell == NULL)
+        return;
+    if (menustr == NULL)
+        return;
+
+    box = cell->cell.gui_private;
+
+    if (box->item_list != NULL)
+    {
+        if (box->register_is_reversed)
+        {
+            box->occurrence_id++;
+            if (g_hash_table_lookup_extended (box->item_hash, menustr, NULL, NULL))
+                return;
+        }
+        else
+            box->occurrence_id--;
+
+        g_hash_table_insert (box->item_hash, g_strdup (menustr), GINT_TO_POINTER(box->occurrence_id));
+    }
+}
+
+void
+gnc_completion_cell_set_value (CompletionCell* cell, const char* str)
+{
+    if (cell == NULL)
+        return;
+    if (str == NULL)
+        return;
+
+    gnc_basic_cell_set_value (&cell->cell, str);
+}
+
+static inline void
+list_store_append (GtkListStore *store, char* string, char* string_norm, gint occurrence_id)
+{
+    GtkTreeIter iter;
+
+    g_return_if_fail (store != NULL);
+    g_return_if_fail (string != NULL);
+    g_return_if_fail (string_norm != NULL);
+
+    gtk_list_store_append (store, &iter);
+    gtk_list_store_set (store, &iter, TEXT_COL, string,
+                                      TEXT_NORMAL_COL, string_norm,
+                                      OCCURANCE_COL, occurrence_id, -1);
+}
+
+static char*
+casefold (char* utf8_string)
+{
+    char *folded = NULL;
+    g_return_val_if_fail (utf8_string && *utf8_string, NULL);
+
+    folded = g_utf8_casefold (utf8_string, -1);
+    return folded;
+}
+
+static char*
+normalize (char* utf8_string)
+{
+    char *normalized = NULL;
+    g_return_val_if_fail (utf8_string && *utf8_string, NULL);
+
+    normalized = g_utf8_normalize (utf8_string, -1, G_NORMALIZE_NFC);
+    return normalized;
+}
+
+static char*
+normalize_and_fold (char* utf8_string)
```

Comment:
This function is only called once. Considering all the other wrapping around g_utf8_... is redundant, you may as well just use the two g_utf8_... calls directly in `populate_list_store`, and `g_free` the intermediate result there.

Reply:
Same as above...